AmplifyPlugins/API/AWSAPICategoryPlugin/Support/Internal/URLComponents+sigv4Encoding.swift

```diff
@@ -22,15 +22,29 @@ extension URLComponents {
         var sigV4UnreservedCharacters = CharacterSet(charactersIn: "A" ... "Z")
         sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "a" ... "z"))
         sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "0" ... "9"))
-        sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "-_~.%=&"))
+        sigV4UnreservedCharacters = sigV4UnreservedCharacters.union(CharacterSet(charactersIn: "-_~."))
         return sigV4UnreservedCharacters
     }()
 
-    mutating func percentEncodeQueryBySigV4Rules() {
-        guard let percentEncodedQuery = self.percentEncodedQuery else {
+    private func encodeQueryParamValueBySigV4Rules(_ value: String) -> String {
+        /// removingPercentEncoding returns `nil` if called on a value
+        /// that hasn't been prior encoded
+        let unencoded = value.removingPercentEncoding ?? value
+
+        /// apply sigv4 encoding
+        return unencoded.addingPercentEncoding(
```

Comment:
1. 🙃
2. I'm not sure about under what conditions `addingPercentEncoding` would return return nil, Apple docs only say _"if the transformation is not possible"_ -- possibly if the UTF-8 encoding fails or similar edge case?  I'll update to throw an error that `constructURL` can catch and eventually re-throws an `APIError`

Reply:
As discussed, since this is an internal extension, I don't object to throwing an APIError directly from the extension, rather than adding a custom type of limited value. Let's just make sure the APIError has enough detail that the customer can diagnose the problem if they encounter the rare case where a URL can't be encoded.